### ISSUE TITLE
Format output from diff differently

### ISSUE THREAD

**Issue by ikwyl6** [odeke-em/drive#89]
I have a hard time trying to decipher what your diff is showing for what files and for what changes. For example:

`$ drive diff Taxes/` gives:

```
* remote:         2015-02-25T23:24:9.000Z                 
* local:          2014-11-15T17:19:56.000Z                
********************************************************************************
ikwyl6 - Netfile Federal 2013.tax ikwyl6 - Netfile Federal 2013.tax
********************************************************************************
********************************************************************************
/Documents/Taxes/2013 Tax Year/ikwyl6 - Netfile Federal 2013.tax only on remote
* remote:         2014-12-28T14:03:14.000Z                
* local:          2013-04-13T00:52:14.000Z                
********************************************************************************
ikwyl6 - Netfile Federal 2012.tax ikwyl6 - Netfile Federal 2012.tax
********************************************************************************
* remote:         2014-12-28T14:01:36.000Z                
* local:          2013-04-13T00:52:17.000Z                
********************************************************************************
ikwyl6 - Tax Return 2012.pdf ikwyl6 - Tax Return 2012.pdf
********************************************************************************
********************************************************************************
/Documents/Taxes/2012 Tax Year/ikwyl6 - Tax Return 2012.pdf only on remote
********************************************************************************
/Documents/Taxes/2012 Tax Year/ikwyl6 - Netfile Federal 2012.tax only on remote
* remote:         2013-03-10T21:35:28.000Z                
* local:          2013-03-10T21:35:15.000Z                
********************************************************************************
UFile - Income Tax 2008, return08h.pdf UFile - Income Tax 2008, return08h.pdf
********************************************************************************
********************************************************************************
/Documents/Taxes/2008 Tax Year/T2202A - 2008taxyear.ps only on remote
********************************************************************************
/Documents/Taxes/2008 Tax Year/UFile, income tax 2008, net08h.tax only on remote
********************************************************************************
/Documents/Taxes/2008 Tax Year/UFile, income tax 2008 CRA confirmation.ps only on remote
********************************************************************************
/Documents/Taxes/2008 Tax Year/UFile - Income Tax 2008, return08h.pdf only on remote
********************************************************************************
/Documents/Taxes/Chiro Receipts (4).pdf only on remote

```

First of all, I get the "only on remote" or "only on local" - that is self explanatory. 

Does the first line tell me about 'Taxes' directory alone? I have no idea.

What about "ikwyl6 - Netfile Federal 2012.tax ikwyl6 - Netfile Federal 2012" and the lines after the **** and remote and local dates? Is this for this file or for "ikwyl6 - Tax Return 2012.pdf ikwyl6 - Tax Return 2012.pdf" or for the one below it?

For me - I think maybe the **** (and double *** lines) are making this a little hard to read and to tell which local/remote dates are for what files. Personally, I would rather you only have 3 or 4 stars between each file, so that way you can tell what it is. And any "only on remote" or "only on local" don't need ****'s between these lines. As well, if you are going to show the timestamps of the files, you should really show the file sizes as well. FYI For this example, I only had these files on Google Drive and not locally and did a pull, but I'm still getting a diff (Shouldn't there be no diff returned?). 

Anyways, for the way I think it should look, here it is:

```
****
Directory: Taxes/
remote:         2015-02-25T23:24:9.000Z                 
local:          2014-11-15T17:19:56.000Z    
remote file size: 4096 bytes
local file size: 4096 bytes
****
File: ikwyl6 - Netfile Federal 2013.tax
remote:         2014-12-28T14:01:36.000Z                
local:          2013-04-13T00:52:17.000Z      
remote file size: 2334226 bytes
local file size: 2334226 bytes
****
/Documents/Taxes/2008 Tax Year/T2202A - 2008taxyear.ps only on remote
/Documents/Taxes/2008 Tax Year/UFile, income tax 2008, net08h.tax only on remote
/Documents/Taxes/2008 Tax Year/UFile, income tax 2008 CRA confirmation.ps only on remote
/Documents/Taxes/2008 Tax Year/UFile - Income Tax 2008, return08h.pdf only on remote
```

**Comment by odeke-em** [comment]
Good plan, thank you for raising this!

**Comment by ikwyl6** [comment]
As well, it would be nice to have an option to just show what directories are different, instead of it's nested files. Or just report that dir1, dir2, dir3 have files within them that are different but not list those files in the command, just the parent directories.

Sometimes the diff output could be very long and sometimes you may want to just look at each 'diffing' folder at a time to make sure things are sync'ed.

**Comment by ikwyl6** [comment]
Any thoughts on implementing this on the next release? I think both remote and local dates and byte sizes should be shown, regardless if the sizes are the same or the dates are the same, but one of the other is (obviously) different.

**Comment by odeke-em** [comment]
Coming soon ;), please ping me up on Wednesday if I haven't yet given you a status on the implementation.

**Comment by l3iggs** [comment]
Maybe the diff output should follow the [GNU Diffutils](https://www.gnu.org/software/diffutils/) unified diff format [as defined here](http://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html). That way it could be used with `patch` and other handy unix utils which would be really cool actually.

This seems like it could be annoying to get right and seems like a "nice to have" feature IMHO.

**Comment by odeke-em** [comment]
Thank you @l3iggs I shall take a look at that.

**Comment by ikwyl6** [comment]
Perhaps my version (short version of diff) above (or something similar) could be the default, but the extensive diffutils version could be a --verbose or --diffutils flag? Showing large diffs by default, even for 1 short file can be tedious if you just want to see shorthand what the changes are. Or vice versa, show long version and have a --short flag.

**Comment by odeke-em** [comment]
Btw if anyone would like to implement this, feel free to do so. This is because the requirements seems to be to quite a specific/strict liking. Otherwise, I started working on this on branch  https://github.com/odeke-em/drive/tree/diff-prettify and PR https://github.com/odeke-em/drive/pull/122

@ikwyl6 for:
```
****
Directory: Taxes/
remote:         2015-02-25T23:24:9.000Z                 
local:          2014-11-15T17:19:56.000Z    
remote file size: 4096 bytes
local file size: 4096 bytes
****
```
We won't be diff-ing directories, only the file inside them. This seems like a thing you'd do with `stat` to get information about folders/directories.


@l3iggs btw just to let you know drive does not perform its own diff. It just exec-s a subprocess to diff on your system with stdout and stderr connected. I had experimented around with different diff-ing go libraries and they did not work seem to work, consumed too much memory, produced bad output. I just gave up since it seemed like a time sink with no progress: possibly until I could find a good go diff library.
